examples/php-xhr/upload.php

```diff
@@ -0,0 +1,35 @@
+<?php
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL);
+
+if($_SERVER["REQUEST_METHOD"] == "OPTIONS")
+{
+    if (isset($_SERVER["HTTP_ACCESS_CONTROL_REQUEST_METHOD"]))
+        header("Access-Control-Allow-Methods: POST, GET, OPTIONS, DELETE, PUT"); //Make sure you remove those you do not want to support
+        header("Access-Control-Allow-Origin: *");
+
+    if (isset($_SERVER["HTTP_ACCESS_CONTROL_REQUEST_HEADERS"]))
+        header("Access-Control-Allow-Headers: {$_SERVER['HTTP_ACCESS_CONTROL_REQUEST_HEADERS']}");
+
+    //Just exit with 200 OK with the above headers for OPTIONS method
+    exit(0);
+}
+
+if($_POST) {
+      $target_dir = "./uploads/";
+      $target_file = $target_dir . basename($_FILES["files"]["name"][0]);
+      if (move_uploaded_file($_FILES["files"]["tmp_name"][0], $target_file)) {
+        header("Access-Control-Allow-Origin: *");
+        header('Content-type: application/json');
+        $data = ["url" => $target_file, "message" => "The file ". basename( $_FILES["files"]["name"][0]). " has been uploaded."];
+        http_response_code(201);
+        echo json_encode( $data );
+      } else {
+        header("Access-Control-Allow-Origin: *");
+        header('Content-type: application/json');
+        $data = ["message" => "Sorry, there was an error uploading your file."];
```

Comment:
More descriptive error would be good

Reply:
Maybe I should use a try catch? and send the error array from the throwable?

Reply:
Not sure if that works? I don't think it throws right? So maybe just sth like: Unable to move the uploaded file to its final location: X. Please check up on server permissions.

Reply:
Yes it throws, using the throwable path. Not sure the `move_uploaded_file` function throws, as it returns a boolean. I can throw an error from the block when the boolean is false;

Reply:
Updated it, I guess this way, if any other error that throws comes up, it is also handled